src/utilities/device_utils.c

```diff
@@ -2085,28 +2088,44 @@ template HYPRE_Int hypreDevice_CsrRowPtrsToIndicesWithRowNum( HYPRE_Int     nrow
 
 /*--------------------------------------------------------------------
  * hypreGPUKernel_DiagScaleVector
+ *
+ * Assumes vectors components are stored contiguously.
  *--------------------------------------------------------------------*/
 
+template <HYPRE_Int NV>
 __global__ void
 hypreGPUKernel_DiagScaleVector( hypre_DeviceItem &item,
-                                HYPRE_Int         n,
-                                HYPRE_Int        *A_i,
-                                HYPRE_Complex    *A_data,
+                                HYPRE_Int         num_vectors,
+                                HYPRE_Int         num_rows,
+								HYPRE_Int        *A_i,
+								HYPRE_Complex    *A_data,
                                 HYPRE_Complex    *x,
                                 HYPRE_Complex     beta,
                                 HYPRE_Complex    *y )
 {
-   HYPRE_Int i = hypre_gpu_get_grid_thread_id<1, 1>(item);
+   HYPRE_Int     i = hypre_gpu_get_grid_thread_id<1, 1>(item);
+   HYPRE_Int     j;
+   HYPRE_Complex val;
 
-   if (i < n)
+   if (i < num_rows)
    {
+      val = 1.0 / A_data[A_i[i]];
+
       if (beta != 0.0)
       {
-         y[i] = x[i] / A_data[A_i[i]] + beta * y[i];
+#pragma unroll NV
+         for (j = 0; j < num_vectors; ++j)
```

Comment:
@PaulMullowney, very minor question: is there a specific reason to use `++j` over `j++`? We prefer the latter in terms of style.

Reply:
I think that's just an old habit of mine. You can change it back to j++